src/gui/Mesh.cpp

```diff
@@ -0,0 +1,228 @@
+                tempIndexArrays.push_back(std::move(index));
+            else
+                throw "VertexData: invalid file!"; // TODO_EXCEPTION
+        }
+
+        while (*buffer++ != 10)
+            if (*buffer == 0)
+                break;
+    }
+
+    std::map<std::array<unsigned, 3>, unsigned> createdVertices;
+
+    bool usingTexCoords = tempTexCoords.size() > 0;
+    bool usingNormals = tempNormals.size() > 0;
+    bool usingIndexing = true;
+
+    for (auto& indexArray : tempIndexArrays) {
+        if (usingTexCoords && (indexArray[3] == 0 || indexArray[4] == 0 || indexArray[5] == 0))
+            throw "VertexData: invalid index data (texture coordinates)";
+
+        if (usingNormals && (indexArray[6] == 0 || indexArray[7] == 0 || indexArray[8] == 0))
+            throw "VertexData: invalid index data (normals)";
+
+        std::array<unsigned, 3> v1 = { indexArray[0], indexArray[3], indexArray[6] };
+        std::array<unsigned, 3> v2 = { indexArray[1], indexArray[4], indexArray[7] };
+        std::array<unsigned, 3> v3 = { indexArray[2], indexArray[5], indexArray[8] };
+
+        if (createdVertices[v1] == 0) {
+            positions.push_back(tempPositions.at(indexArray[0]-1));
+            if (usingTexCoords)
+                texCoords.push_back(tempTexCoords.at(indexArray[3]-1));
+            if (usingNormals)
+                normals.push_back(tempNormals.at(indexArray[6]-1));
+
+            createdVertices[v1] = positions.size()-1;
+        }
+        indices.push_back(createdVertices[v1]);
+
+        if (createdVertices[v2] == 0) {
+            positions.push_back(tempPositions.at(indexArray[1]-1));
+            if (usingTexCoords)
+                texCoords.push_back(tempTexCoords.at(indexArray[4]-1));
+            if (usingNormals)
+                normals.push_back(tempNormals.at(indexArray[7]-1));
+
+            createdVertices[v2] = positions.size()-1;
+        }
+        indices.push_back(createdVertices[v2]);
+
+        if (createdVertices[v3] == 0) {
+            positions.push_back(tempPositions.at(indexArray[2]-1));
+            if (usingTexCoords)
+                texCoords.push_back(tempTexCoords.at(indexArray[5]-1));
+            if (usingNormals)
+                normals.push_back(tempNormals.at(indexArray[8]-1));
+
+            createdVertices[v3] = positions.size()-1;
+        }
+        indices.push_back(createdVertices[v3]);
+    }
+
+    //  create and bind the VAO
+    glGenVertexArrays(1, &vertexArrayObjectId_);
+    glBindVertexArray(vertexArrayObjectId_);
+
+    //  upload the vertex data to GPU and set up the vertex attribute arrays
+    glGenBuffers(1, &positionBufferId_);
+    glBindBuffer(GL_ARRAY_BUFFER, positionBufferId_);
+    glBufferData(GL_ARRAY_BUFFER, positions.size() * sizeof(std::array<float, 4>), &positions[0], GL_STATIC_DRAW);
+    glEnableVertexAttribArray(0);
+    glVertexAttribPointer(0, 4, GL_FLOAT, GL_FALSE, 0, (GLvoid*)0);
+/*
```

Comment:
Should this be enabled or removed (probably with rest of the texture coordinate parsing)?